Release anxiety blog post (part 1) (#4959)
sourcegraph/about: blogposts/release-anxiety.md

---
title: "Let's talk about release anxiety"
description: 'Engineers are caught in a vise between two pressures: the pressure to get code changes into the next release and to protect the release from bugs. The result? Release anxiety.'
externalTitle: 'Release anxiety: What it is and how to address it'

> externalDescription: 'Release anxiety plagues many engineers. Can you squeeze in this last code change? Will this last code change break production? Let's talk about ways to deal with the anxiety.'

Reviewer: ```suggestion
externalDescription: "Release anxiety plagues many engineers. Can you squeeze in this last code change? Will this last code change break production? Let's talk about ways to deal with the anxiety."
```

author: Nick Moore
authorUrl: https://twitter.com/nickwritesit
published: true
publishDate: 2021-11-04T10:00-07:00
tags: [blog]
slug: release-anxiety
heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/release-anxiety-images/release-anxiety-hero.png
socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/release-anxiety-images/release-anxiety-hero.png
---

> ![Release anxiety graphic]https://storage.googleapis.com/sourcegraph-assets/blog/release-anxiety-images/release-anxiety-hero.png

Reviewer 2: Image isn't rendering properly 
<img width="649" alt="Screen Shot 2021-11-03 at 8 50 53 AM" src="https://user-images.githubusercontent.com/73327004/140083642-7de0c29b-ce22-4151-8e17-67f142fbd04e.png">
g

> “Don’t release to prod on Fridays”–a saying that might as well be sticky-noted to every engineer’s monitor. It’s a joke that masks something pretty serious: engineers don’t feel confident in their ability to deploy and walk away from their desks.

Reviewer 2: ```suggestion
“Don’t release to prod on Fridays” is a saying that might as well be sticky-noted to every engineer’s monitor. It’s a joke that masks something pretty serious: engineers don’t feel confident in their ability to deploy and walk away from their desks.
```

Style choice nitpick

The authors of _[The Phoenix Project](https://www.amazon.com/Phoenix-Project-DevOps-Helping-Business/dp/1942788290/)_ compared this fear to the instinct animals have to avoid danger, writing that “everyone is moving their changes as far away from Friday as possible, like woodland creatures running away from a forest fire.”

But our workplaces should, ideally, never resemble forest fires. The result of impending fire is release anxiety–when engineers feel vigilant, nervous, and wary of upcoming releases and the release process in general.

> Sourcegraph, at least in this way, is not unique. Our engineers get anxious, too (one of our engineers, [Adam Harvey](https://handbook.sourcegraph.com/company/team#adam-harvey-hehim), wrote an entire [RFC](https://handbook.sourcegraph.com/communication/rfcs) on the subject). Where we are unique, however, is our concurrent release strategies and our methods for handling them. For our self-hosted product, we have a monthly release schedule; for our cloud product on sourcegraph.com, we release continuously.

Reviewer 2: ```suggestion
Sourcegraph, at least in this way, is not unique. Our engineers get anxious, too (one of our engineers, [Adam Harvey](https://handbook.sourcegraph.com/company/team#adam-harvey-hehim), wrote an entire [RFC](https://handbook.sourcegraph.com/communication/rfcs) on the subject). Where we are unique, however, is our concurrent release strategies and our methods for handling them. For our [self-hosted product](https://about.sourcegraph.com/#get-started), we have a monthly release schedule; for our cloud product on [sourcegraph.com](https://sourcegraph.com/), we release continuously.
```

Add product links to references

And if you’ve felt more pressure lately than you used to, you’re not alone. According to our [Big code report](https://info.sourcegraph.com/emergence-of-big-code-2020-survey?_ga=2.117030534.1813537383.1623859682-2046487521.1621427028), 92% of participants reported that the pressure they feel to release code faster has increased in the past 10 years. And there’s no sign of this pressure decreasing.

### The pressure to improve

In software, especially if you’re in a startup environment, there’s a constant push to grow and show value. To stand still is to atrophy and/or risk disruption.

This pressure comes both from the external expectations of your market and your customers as well as internal expectations from your company’s leadership. Often, your ability to develop and submit code changes is one of your, or one of your team’s, OKRs, meaning your [career progression](https://about.sourcegraph.com/blog/software-engineer-career-ladder/) can become dependent, in part, on the release cycle.

> Downstream, that push results in a pressure applied to every release. Release anxiety is likely to get out of hand if every release carries the weight of the product’s future. In one sense, this anxiety is neurotic–one release isn’t going to kill your company; in another sense, it’s all too real–an important customer may very well be on the other end of a release, waiting for the change they asked for.

Reviewer 2: ```suggestion
Downstream, that push results in a pressure applied to every release. Release anxiety is likely to get out of hand if every release carries the weight of the product’s future. In one sense, this anxiety is neurotic–one release isn’t going to kill your company. In another sense, it’s all too real–an important customer may very well be on the other end of a release, waiting for the change they asked for.
```
Separate into two sentences to simplify the sentence structure

---

> title: 'Why the antidote to release anxiety is a continuous delivery mindset'

Author: @rebeccadee I like that idea! I think there are two distinct ideas here, too, so they could benefit from some separation. I'll work up a second PR for your review.

description: 'Release anxiety cannott be cured but you can alleviate it. Learn how a continuous delivery mindset can help you break features into smaller iterations and keep your code in a secure and releaseable state.'
author: Nick Moore
authorUrl: https://twitter.com/nickwritesit
published: true
publishDate: 2021-10-28T10:00-07:00
tags: [blog]
slug: release-anxiety
heroImage:
socialImage:
---

“Don’t release to prod on Fridays”–a saying that might as well be sticky-noted to every engineer’s monitor. It’s a joke that masks something pretty serious: engineers don’t feel confident in their ability to deploy and walk away from their desks.

> The authors of \_[The Phoenix Project](https://www.amazon.com/Phoenix-Project-DevOps-Helping-Business/dp/1942788290/ref=asc_df_1942788290/) \_compared this fear to the instinct animals have to avoid danger, writing that “everyone is moving their changes as far away from Friday as possible, like woodland creatures running away from a forest fire.”

Author: I wish I knew! I keep deleting them and then they come back 😅

title: "Let's talk about release anxiety"
description: 'Engineers are caught in a vise between two pressures: the pressure to get code changes into the next release and to protect the release from bugs. The result? Release anxiety.'
externalTitle: 'Release anxiety: What it is and how to address it'

> externalDescription: "Release anxiety plagues many engineers. Can you squeeze in this last code change? Will this last code change break production? Let's talk about ways to deal with the anxiety."

Reviewer: @nickmyyz We've seen this issue before – there was an apostrophe in this line causing the build to break because it didn't have a corresponding one. I've fixed it by enclosing everything in double quotation marks!

Author: @rebeccadee dang, I was close! I fixed that in the title but didn't see the apostrophe in the external. Thank you! :)
